### PR TITLE
chore(dev): Setup webpack-dev-server with HMR

### PR DESCRIPTION
**Pull request by bdauria**
## Description
This PR introduces the webpack-dev-server with HMR and react-hot-loader. This improves greatly the development workflow by reloading modules on the fly when something changes instead of having to reload the whole page.

HMR is not supported for external modules yet as it's not compatible with System js at the moment. We might need to wait for webpack 5 to be able to build external ES modules or give https://github.com/alexisvincent/systemjs-hot-reloader a try

**This PR should be merged after final 20.04 build**

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x (master)
- [x] 20.10.x (next)

### CODE REVIEW

File: webpack.config.dev.js [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 const path = require('path');
 
 const devConfig = require('@centreon/frontend-core/webpack/patch/dev');
 const baseConfig = require('./webpack.config');
 
+const devServerPort = 9090;
+
 module.exports = merge(baseConfig, devConfig, {
+  output: {
+    publicPath: `http://localhost:${devServerPort}/static/`,
```
Comment (kduret) [inline webpack.config.dev.js:11]: Does it work if browser is not on the same machine than the build

File: package.json [hunk 2]
```diff
@@ -90,6 +92,7 @@
     "filesize": "^6.1.0",
     "formik": "^2.1.4",
     "html-react-parser": "^0.10.3",
+    "html-webpack-harddisk-plugin": "^1.0.1",
```
Comment (kduret) [inline package.json:95]: should be a dev dependency

File: webpack.config.dev.js [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
+  },
   resolve: {
     alias: {
       react: path.resolve('./node_modules/react'),
+      'react-dom': '@hot-loader/react-dom',
       'react-router-dom': path.resolve('./node_modules/react-router-dom'),
       '@material-ui/core': path.resolve('./node_modules/@material-ui/core'),
     },
   },
-  plugins: [new LiveReloadPlugin({ appendScriptTag: true })],
```
Comment (kduret) [inline webpack.config.dev.js:16]: must be removed from package.json

File: webpack.config.dev.js (hunk 1 above)
Comment (bdauria) [reply webpack.config.dev.js:11]: It works even if the index.html file is served remotely, but the dev server must run locally.

File: webpack.config.js [hunk 4]
```diff
@@ -20,9 +25,11 @@ module.exports = merge(baseConfig, extractCssConfig, {
   },
   plugins: [
     new HtmlWebpackPlugin({
+      alwaysWriteToDisk: true,
```
Comment (jdelpierre) [inline webpack.config.js:28]: this is for a better performance... or something else?

Comment (bdauria) [reply webpack.config.js:28]: We need to write the HTML file on disk, not just serve it. It's needed by the docker container which is already serving the file.

**Comment by Thebarda** [comment]
Tested with latest commit and LGTM. 
Build time roughly increased maybe two times less 😍

**Pull request merged**